packages/dataviews/src/types.ts

```diff
@@ -525,37 +525,41 @@ export interface SupportedLayouts {
 	table?: Omit< ViewTable, 'type' >;
 }
 
-export interface CombinedFormField< Item > extends CombinedField {
-	render?: ComponentType< { item: Item } >;
+interface BaseFieldLayout {
+	id: string;
+	label?: string;
 }
 
-export interface DataFormCombinedEditProps< Item > {
-	field: NormalizedCombinedFormField< Item >;
-	data: Item;
-	onChange: ( value: Record< string, any > ) => void;
-	hideLabelFromVision?: boolean;
+export interface RegularFieldLayout extends BaseFieldLayout {
+	layout: 'regular';
 }
 
-export type NormalizedCombinedFormField< Item > = CombinedFormField< Item > & {
-	fields: NormalizedField< Item >[];
-	Edit?: ComponentType< DataFormCombinedEditProps< Item > >;
-};
+export interface PanelFieldLayout extends BaseFieldLayout {
+	layout: 'panel';
+	fields?: FormField[];
+}
+
+export interface InlineFieldLayout extends BaseFieldLayout {
+	layout: 'inline';
+}
+
+export type FormField =
+	| string
+	| RegularFieldLayout
+	| PanelFieldLayout
+	| InlineFieldLayout;
```

Comment:
What's the difference between RegularField and InlineField?

Reply:
Regular field:
![Screenshot 2024-11-04 at 1 27 05 PM](https://github.com/user-attachments/assets/2d3cfb21-4cc6-4f55-b0cb-35508df7e3ed)

Inline field:
![Screenshot 2024-11-04 at 1 27 18 PM](https://github.com/user-attachments/assets/7a0e9725-e2b5-42c6-9890-07497dd94aeb)

Inline is in the column layout similar to panel field, but displays the actual edit field on the right side.
Happy to change the name if there is a better alternative?

Reply:
Here's my take on this, I think we only have two kind of fields: `inline` (or regular) and `panel` and both can support labelPosition as "side" or "top". It's kind of similar to the "labelPosition" prop in React Spectrum https://react-spectrum.adobe.com/react-spectrum/TextField.html#label-alignment-and-position

Reply:
👍 I like that idea as well, lets go with that.
I will just have to see how the `top` label position will look like for the panel view, given just moving the button below the label may be a bit strange:
![Screenshot 2024-11-04 at 2 50 42 PM](https://github.com/user-attachments/assets/a9a14eaf-1d18-4339-81dd-50fb8a704f5c)

I guess wrapping it with the `BaseControl` could suffice:
![Screenshot 2024-11-04 at 2 52 08 PM](https://github.com/user-attachments/assets/de40c5c8-7ff7-48a0-88f0-7bb6670c4c40)

Reply:
If the "button" is labeled, Adding `BaseControl` might create two labels (so not entirely correct a11y wise), so maybe the visuals should match what you shared above but it should just be adhoc matching.